src/shadowbox/server/manager_service.ts

```diff
@@ -248,26 +252,23 @@ export class ShadowsocksManagerService {
   public async setAccessKeyDataLimit(req: RequestType, res: ResponseType, next: restify.Next) {
     try {
       logging.debug(`setAccessKeyDataLimit request ${JSON.stringify(req.params)}`);
-      const accessKeyId = validateAccessKeyId(req.params.id);
-      const limit = req.params.limit as DataUsage;
-      if (!limit) {
+      const limit = req.params.limit as DataLimit;
+      if (limit === undefined) {
         return next(
-            new restify.MissingParameterError({statusCode: 400}, 'Parameter `limit` is missing'));
-      } else if (!Number.isInteger(limit.bytes)) {
-        return next(new restify.InvalidArgumentError(
-            {statusCode: 400}, 'Parameter `limit.bytes` must be an integer'));
+            new restify.MissingParameterError({statusCode: 400}, 'Missing `limit` parameter'));
+      } else if (!!limit && !Number.isInteger(limit.bytes)) {
```

Comment:
`!Number.isInteger(limit.bytes) || limit.bytes >= 0`?

Reply:
The repo validates the actual value. I simplified this to `!Number.isInteger(limit.bytes)`.